app/src/views/OperationalLearning/Stats/index.tsx

```diff
@@ -0,0 +1,352 @@
+import useAlert from '#hooks/useAlert';
+import {
+    type GoApiResponse,
+    type GoApiUrlQuery,
+    useRequest,
+} from '#utils/restRequest';
+
+import OperationalLearningMap from './OperationalLearningMap';
+
+import i18n from './i18n.json';
+import styles from './styles.module.css';
+
+type OpsLearningQuery = GoApiUrlQuery<'/api/v2/ops-learning/'>;
+type OpsLearningSummaryResponse = GoApiResponse<'/api/v2/ops-learning/stats/'>;
+type SectorStatItem = NonNullable<OpsLearningSummaryResponse['learning_by_sector']>[number];
+type RegionStatItem = NonNullable<OpsLearningSummaryResponse['learning_by_region']>[number];
+type SourcesOverTimeItem = NonNullable<OpsLearningSummaryResponse['sources_overtime']>[number];
+
+const sectorKeySelector = (datum: SectorStatItem) => datum.sector_id;
+const sectorValueSelector = (datum: SectorStatItem) => datum.count;
+const sectorLabelSelector = (datum: SectorStatItem) => datum.title;
+
+const regionKeySelector = (datum: RegionStatItem) => datum.region_id;
+const regionValueSelector = (datum: RegionStatItem) => datum.count;
+const regionLabelSelector = (datum: RegionStatItem) => datum.region_name;
+
+type SourceType = 'dref' | 'emergencyAppeal' | 'others';
+interface SourceTypeOption {
+    key: SourceType;
+    label: string;
+    color: string;
+}
+
+const dataKeys: SourceType[] = [
+    'dref',
+    'emergencyAppeal',
+    'others',
+];
+const dataKeyToClassNameMap: Record<SourceType, string> = {
+    dref: styles.dref,
+    emergencyAppeal: styles.emergencyAppeal,
+    others: styles.others,
+};
+const sourceClassNameSelector = (dataKey: SourceType) => dataKeyToClassNameMap[dataKey];
+const xAxisFormatter = (date: Date) => date.toLocaleString(
+    navigator.language,
+    { year: 'numeric' },
+);
+
+type SourceTypeEnum = components<'read'>['schemas']['ApiAppealTypeEnumKey'];
+const SOURCE_TYPE_EMERGENCY = 1 satisfies SourceTypeEnum;
+const SOURCE_TYPE_DREF = 0 satisfies SourceTypeEnum;
+
+const transformSourcesOverTimeData = (data: SourcesOverTimeItem[]) => {
+    const groupedData: Record<string, Record<SourceType, number>> = {};
+
+    data.forEach((entry) => {
+        const year = getFormattedDateKey(entry.date);
+        if (!groupedData[year]) {
+            groupedData[year] = { dref: 0, emergencyAppeal: 0, others: 0 };
+        }
+        if (entry.atype === SOURCE_TYPE_DREF) {
+            groupedData[year].dref += entry.count;
+        } else if (entry.atype === SOURCE_TYPE_EMERGENCY) {
+            groupedData[year].emergencyAppeal += entry.count;
+        } else {
+            groupedData[year].others += entry.count;
+        }
+    });
+
+    return groupedData;
+};
+
+interface Props {
+    query: OpsLearningQuery | undefined
+}
+
+function Stats(props: Props) {
+    const {
+        query,
+    } = props;
+
+    const strings = useTranslation(i18n);
+    const alert = useAlert();
+    const [activePointKey, setActivePointKey] = useState<string>();
```

Comment:
Is this type correct? The state is not defined here.

Reply:
Yes the type here is correct.